src/Workspaces/Core/Portable/ExternalAccess/EditorConfig/EditorConfigGenerator.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Options;
+
+namespace Microsoft.CodeAnalysis.ExternalAccess.EditorConfig
+{
+    [Export(typeof(EditorConfigGenerator)), Shared]
+    internal class EditorConfigGenerator
+
+    {
+        private readonly IGlobalOptionService _globalOptions;
+        private readonly ImmutableArray<Lazy<IEditorConfigOptionsCollection, LanguageMetadata>> _editorConfigGenerators;
+
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public EditorConfigGenerator(
+            IGlobalOptionService globalOptions,
+            [ImportMany] IEnumerable<Lazy<IEditorConfigOptionsCollection, LanguageMetadata>> generators)
+        {
+            _globalOptions = globalOptions;
+            _editorConfigGenerators = generators.ToImmutableArray();
+        }
+
+        public string? Generate(string language)
+        {
+            var groupOptions = GetDefaultOptions(language);
+            return EditorConfigFileGenerator.Generate(groupOptions, _globalOptions, language);
+        }
+
+        public ImmutableArray<(string feature, ImmutableArray<IOption2> options)> GetDefaultOptions(string language)
+        {
+            var generators = _editorConfigGenerators.Where(b => b.Metadata.Language == language);
```

Comment:
@tmat You gave the feedback that this should be an array. Do you have more context on how the .editorconfig file should work for multiple languages? I'm assuming if our solution has both vb and cs projects the editorconfig should include both. The existing [file generate ](https://github.com/dotnet/roslyn/blob/73b525f7ffdb8dae56e148ccadd8385ce626fdb4/src/Workspaces/Core/Portable/Options/EditorConfig/EditorConfigFileGenerator.cs#L18)method only allows one language. I tried to reproduce in 2019 to see the behavior, but it didn't include both languages. 

I can make this an array and append the additional rules to the end, I'm just not sure if that is how this is supposed to work.